src/button.scss

```diff
@@ -11,456 +45,454 @@
   }
 }
 
-@mixin focus() {
-  outline: var(--sapContent_FocusWidth) dotted var(--sapContent_FocusColor);
-  outline-offset: -0.1875rem;
-}
-
-@mixin standard() {
-  color: var(--sapButton_TextColor);
-  border-color: var(--sapButton_BorderColor);
-  background-color: var(--sapButton_Background);
-}
-
-@mixin positive() {
-  color: var(--sapButton_Accept_TextColor);
-  border-color: var(--sapButton_Accept_BorderColor);
-  background-color: var(--sapButton_Accept_Background);
+@mixin iconSize($size) {
+  &::before,
+  &::after {
+    font-size: $size;
+    line-height: $size;
+  }
 }
 
-@mixin negative() {
-  color: var(--sapButton_Reject_TextColor);
-  border-color: var(--sapButton_Reject_BorderColor);
-  background-color: var(--sapButton_Reject_Background);
-}
+@mixin buttonBase() {
+  height: $fd-button-height;
+  max-height: $fd-button-height;
+  min-width: $fd-button-min-width;
+  line-height: $fd-button-line-height;
 
-@mixin isStandard() {
-  @include standard();
+  @include iconSize($fd-button-icon-font-size);
 
-  @include fd-hover() {
-    color: var(--sapButton_Hover_TextColor);
-    border-color: var(--sapButton_Hover_BorderColor);
-    background-color: var(--sapButton_Hover_Background);
-  }
+  padding-left: $fd-button-padding-x;
+  padding-right: $fd-button-padding-x;
+  text-align: center;
 
-  @include fd-disabled() {
-    @include isDisabled();
-    @include standard();
+  // look
+  border-style: solid;
+  border-width: $fd-button-border-width;
 
-    &:focus {
-      outline: none;
-    }
+  // animation
+  transition: all $fd-button-transition-params;
 
-    box-shadow: none;
+  // placements
+  &::before {
+    margin-right: $fd-button-icon-margin;
+    vertical-align: text-bottom;
   }
 
-  @include fd-active-pressed-selected() {
-    color: var(--sapButton_Selected_TextColor);
-    background-color: var(--sapButton_Active_Background);
-    border-color: var(--sapButton_BorderColor);
+  &::after {
+    margin-left: $fd-button-icon-margin;
+    vertical-align: middle;
+  }
 
-    &:focus {
-      outline-color: var(--sapContent_ContrastFocusColor);
+  @at-root {
```

Comment:
I might be wrong, but this code will globally affect all:
- `fd-button[dir="rtl"]::before` 
- `[dir="rtl"]fd-button::before`

Since this behavior is still limited by association with `fd-button` class selector what is the point of registering it `@at-root`?

Reply:
from what I know it will only affect button, although this is a leftover

Reply:
fixed